core/backend/collection.go

```diff
@@ -129,6 +129,11 @@ func (b *Backend) createCollectionResource(router *mux.Router, rc collectionConf
 	createIndicesQuery := fmt.Sprintf("CREATE index IF NOT EXISTS %s ON %s.\"%s\"(timestamp);",
 		"sort_index_"+primary+"_timestamp",
 		schema, resource)
+	if len(foreignColumns) > 0 {
+		createIndicesQuery += fmt.Sprintf("CREATE index IF NOT EXISTS %s ON %s.\"%s\"(%s,timestamp);",
+			"sort_index_"+primary+"_"+strings.Join(foreignColumns, "_")+"_timestamp",
+			schema, resource, strings.Join(foreignColumns, ","))
```

Comment:
We should add these indexies only for the columns that we expect to see together in WHERE clause often. 

If we create an index of (foo_id, bar_id, baz_id, timestamp) it won't be really useful for us if most of our queries are like `... WHERE baz_id = ? AND timestamp > ?;`, i.e. without `foo_id` and `bar_id` 


I don't know how many `foreignColumns` we usually get for a collection, but i think it makes more sense to do individual index for each `foreignColumn` elem, i.e.:

```sql
CREATE index IF NOT EXIST sort_index_primary_foo_id_timestamp ON public.table(foo_id, timestamp);
CREATE index IF NOT EXIST sort_index_primary_bar_id_timestamp ON public.table(bar_id, timestamp);
...
```

Reply:
https://use-the-index-luke.com/sql/where-clause/the-equals-operator/concatenated-keys

Reply:
The number of foreign columns in our current backend is at max 1. If we have more, say A and B, any query by the app will have both A and B defined. However, I agree it could theoretical make sense to have an index with both A and B and one with only B. But since we don't use this, I don't really care.

But the code was still wrong. We sometimes have singletons in subcollections, for example athlete/activity/analysis. Analysis has a timestamp, but no analysis_id, because every analysis object is bound to an activity object. Now any query on analysis was rather fast, because there was an index (athlete_id, activity_id) for the analysis table, but still the database had to sort all analysis rows in memory to get the order right. 

With the updated pull request, we also create an index sort_index_analysis_athlete_id_timestamp(athlete_id,timestamp). This index actually gets used by the backend (verified), so we saved even more CPU.